### DISCUSSION TITLE
Open Files and Folders with StorageProvider

### DISCUSSION THREAD

**Discussion by SirZeeno** [AvaloniaUI/Avalonia#15714] [Q&A]
Is there an implemented option where I am able to open files or folders using the StorageProvider instead of having to choose just one for a file opening prompt? Because I am trying to have a user open a file or folder with just a single button press which then gets imported but, I am only seeing the options to either open a folder selection window or a file selection window, not both.

**Comment by timunie** [comment]
No, you would need two buttons or create your own Managed File/Folder chooser

**Reply by SirZeeno** [reply]
how would I go about making my own one and have it fit into Avalonia functionality and retain multi-platform functionality?

**Reply by timunie** [reply]
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Dialogs/ManagedFileChooser.cs

**Reply by SirZeeno** [reply]
Sorry for the late response, it's been a rough few days. From the link you sent me, can I assume that I would just need to add a OpenFileOrFolderPickerAsync in https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Dialogs/ManagedStorageProvider.cs
that handles opening files or folders depending on what's selected, or are there dependencies that I have to create in the way the function is being handled and so it can get accessed by anyone using Avalonia? because if I make those changes for myself, I would like to make a contribution to the source to add that functionality for everyone.

**Reply by timunie** [reply]
Not sure as this is not intuitive. Most Pickers I know from other tools also don't allow this. So I prefer the current impl in main Avalonia for less confusion. I have simply two buttons where I allow files or folders, but the user has to decide which one they want to ad..

**Reply by thevortexcloud** [reply]
Most platforms treat files and folders as different things. It does not make much sense to allow picking both from the same picker.

**Reply by SirZeeno** [reply]
Like i said, i am not about to treat files and folders as the same and I am not allowing for files and folder to be opened at once, all I would be doing is just make the two buttons that are needed to select either files or folders into one where you can do the exact same thing but without adding complexity and more buttons for the user. under the hood it should basically just work the same way as well but just with checks if you are working with a file or with a directory. All I am trying to do is make the process more simple for the user and more versatile for the developer if they choose to do it that way since this would not replace the standard way of splitting up files and folder selection if your application requires only one of them.
